components/brave_wallet/browser/permission_utils.cc

```diff
@@ -16,7 +16,8 @@
 
 namespace {
 
-constexpr char kAddrPattern[] = "addr%3D(0x[[:xdigit:]]{40})";
+constexpr char kEthAddrPattern[] = "addr%3D(0x[[:xdigit:]]{40})";
+constexpr char kAddrPattern[] = "addr%3D([[:alnum:]]+)";
```

Comment:
Note that these request origins are crafted by us(see GetConcatOriginFromWalletAddresses) and we extract addresses from them. Also we appended these address at the end of host or the origin, so if you can give me a valid example that someone can use a really long and malformed TLD trying to mess with this, then I will add maximum length check. Otherwise I don't think that is necessary.

Reply:
Sounds good I'll try and write some additional tests to see if I can come up with anything

Reply:
@kdenhartog re2 is actually an exception to the rule of 2 as long as the pattern is relatively simple https://chromium.googlesource.com/chromium/src/+/master/docs/security/rule-of-2.md#safe-languages

Reply:
that being said, kEthAddrPattern does as least limit the match to 40 characters and we should probably do something similar in `kAddrPattern`? and also @darkdh shouldn't it be `kSolanaAddrPattern`?

Reply:
> re2 is actually an exception to the rule of 2 as long as the pattern is relatively simple

I hadn't thought about that factor, but yeah I think the grammars here are simple enough that it wouldn't be an issue here. 

> that being said, kEthAddrPattern does as least limit the match to 40 characters and we should probably do something similar in kAddrPattern? and also @darkdh shouldn't it be kSolanaAddrPattern?

Yup that's where I was thinking it would go as well because I was concerned that the process could hang for very long periods of time if there's not a length limit set here. I just figured I'd take the opportunity to write some tests to help out as well as it seemed like a good thing for me to gain better exposure into writing code in this codebase so that I can read it better during reviews too

Reply:
@bridiver I was planning kAddrPattern for other coins not just Solana to make it generic. As I said bumping 40 characters is meaningless unless you can show me the "origin" coming from the website can break our re2 matching on functionality or complexity.

Reply:
see `components/brave_wallet/browser/permission_utils_unittest.cc` to understand how we constructed the request origin. It's either `https://test.com__BrG44HdsEhzapvs8bEqzvkq4egwevS3fRE6ze2ENo6S8` sub-request or `https://test.com{addr=BrG44HdsEhzapvs8bEqzvkq4egwevS3fRE6ze2ENo6S8&addr=JDqrvDz8d8tFCADashbUKQDKfJZFobNy13ugN65t1wvV}` for grouped request.
If you can get a site to host on ex. `test.com.*` which has extremely long TLD, or `test.com_________` which TLD contains bunch of our `__` separator, then we revisit our re2 pattern

Reply:
FWIW, I do have test for separator in domain
https://github.com/brave/brave-core/pull/12945/files#diff-20dbb0e58231ce9287956cc06205d74b1a59432310a597f3e308420fd67e6740R130-R148
and it won't break sub request extraction

Reply:
I will add a comment saying keeping `kEthAddrPattern` is for backward compatibility with existing written content settings, new added coins will always use generic pattern matching

Reply:
Tested GetConcatOriginFromWalletAddresses:
   - with >4kb strings for the domain name
   - with >4kb strings for the addresses name
   - with non-valid URL characters
   - with new line characters
   - with percent encoded characters

No impact for any of these

Testing ParseRequestingOriginFromSubRequest

Hitting some compilation errors at the moment but have tests for the following
   - ETH address too long
   - ETH address too long with = character at end
   - ETH address with = char at the end

Reply:
@kdenhartog those tests are not valid. First you can't get a real world valid origin like those, secondly we won't be able to get those kind of addresses from our wallet.

Reply:
This request origin process would be 
1. We receive ETH permission request from `https://github.com/brave/brave-core/pull/12945`
2. We rewritten the request origin to be https://github.com0xaf5Ad1E10926C0Ee4af4eDAC61DD60E853753f8A in our permission system
3. When we process the request, we extract the address from step 2 request permission
4. If the permission is granted, https://github.com0xaf5Ad1E10926C0Ee4af4eDAC61DD60E853753f8A will be written in content settings

Reply:
Yeah I realized that, but I was considering this to be more about relying on a defensive posture in the off chance that someone is able to manually inject or override an account value that's stored since I didn't want to go digging through that.

Out of curiosity what's the motivation for moving from a 40 character limit that we originally had to an unlimited character limit now? Maybe you're considering a case that I haven't thought about yet.

Reply:
that `kEthAddrPattern` is for backward compatibility for those content settings we already written before this PR, 
new pattern is designed to be **generic** for all types of address I will support in the future, not just Solana

Reply:
Also it's worth pointing out that I understand this issue is probably a bit frustrating since your assumptions do seem very sound at this point and are unlikely to occur. At worst this leads to a DoS attack when someone is doing something that they shouldn't be so the risk of this is quite low and the impact is even lower. 

Now that I think about it from that perspective this probably isn't something worth holding this PR up on any longer since the impact and risk are so low. I'll go ahead and approve it.

Reply:
I've approved so this isn't really a blocking issue

> new pattern is designed to be generic for all types of address we will support in the future, not just Solana

Do you know of any account character length that exists beyond the limit of say 1024 characters?

Reply:
> I've approved so this isn't really a blocking issue
> 
> > new pattern is designed to be generic for all types of address we will support in the future, not just Solana
> 
> Do you know of any account character length that exists beyond the limit of say 1024 characters?

That not I know of but it would be impractical for a very long address

Reply:
128 max length check added and squashed in a4a3e70f83b0171209024c0e7b9af6a9cd3272de